lib/OpenQA/Worker/Cache/Service.pm

```diff
@@ -0,0 +1,206 @@
+
+sub _exists { !!(defined $_[0] && exists $_[0]->{total} && $_[0]->{total} > 0) }
+
+sub active { !app->minion->lock(shift, 0) }
+
+sub enqueued {
+    my $lock = shift;
+    !!($enqueued->grep(sub { $_ eq $lock })->size == 1);
+}
+
+sub dequeue {
+    my $lock = shift;
+    $enqueued = $enqueued->grep(sub { $_ ne $lock });
+}
+
+sub enqueue {
+    push @$enqueued, shift;
+}
+
+sub _setup_workers {
+    return @_ unless grep { /worker/i } @_;
+
+    require OpenQA::Worker::Common;
+    my @args = @_;
+    my ($worker_settings, undef) = OpenQA::Worker::Common::read_worker_config(undef, undef);
+    my $cache_workers
+      = exists $worker_settings->{CACHEWORKERS} ? $worker_settings->{CACHEWORKERS} : DEFAULT_MINION_WORKERS;
+    push(@args, '-j', $cache_workers);
+    return @args;
+}
+
+sub run {
+    shift;
+    require OpenQA::Utils;
+    app->log->short(1);
+    OpenQA::Utils::set_listen_address(7844);
+    $ENV{MOJO_INACTIVITY_TIMEOUT} = 300;
+    require Mojolicious::Commands;
+    my @args = _setup_workers(@_);
+    app->log->debug("Starting cache service: $0 @args");
+    Mojolicious::Commands->start_app('OpenQA::Worker::Cache::Service', @args);
+}
+
+get '/session_token' => sub { shift->render(json => {session_token => SESSION_TOKEN()}) };
+
+get '/info' => sub { $_[0]->render(json => shift->minion->stats) };
+
+post '/status' => sub {
+    my $c    = shift;
+    my $data = $c->req->json;
+    my $lock = _gen_guard_name($data->{lock});
+    my $j    = $data->{id} ? app->minion->job($data->{id})->info : ();
+
+    $c->render(
+        json => {
+            status => (
+                  active($lock)   ? STATUS_DOWNLOADING
+                : enqueued($lock) ? STATUS_IGNORE
+                :                   STATUS_PROCESSED
+            ),
+            (result => $j->{result}, output => $j->{notes}->{output}) x !!($j)});
+};
+
+post '/execute_task' => sub {
+    my $c    = shift;
+    my $data = $c->req->json;
+    my $task = $data->{task};
+    my $args = $data->{args};
+
+    return $c->render(json => {status => STATUS_ERROR, error => 'No Task defined'})
+      unless defined $task;
+    return $c->render(json => {status => STATUS_ERROR, error => 'No Arguments defined'})
+      unless defined $args;
+
+    my $lock = _gen_guard_name($data->{lock} ? $data->{lock} : @$args);
+    app->log->debug("Requested [$task] Args: @{$args} Lock: $lock");
+
+    return $c->render(json => {status => STATUS_DOWNLOADING()}) if active($lock);
+    return $c->render(json => {status => STATUS_IGNORE()})      if enqueued($lock);
+
+    enqueue($lock);
+    my $id = $c->minion->enqueue($task => $args);
+
+    $c->render(json => {status => STATUS_ENQUEUED, id => $id});
+};
+
+get '/' => sub { shift->redirect_to('/minion') };
+
+post '/dequeue' => sub {
+    my $c    = shift;
+    my $data = $c->req->json;
+    dequeue(_gen_guard_name($data->{lock}));
+    $c->render(json => {status => STATUS_PROCESSED});
+};
+
+app->minion->backend->reset;
```

Comment:
`app->minion->reset` works too.

Reply:
Done in https://github.com/os-autoinst/openQA/pull/1783/commits/71e142c959efcf6c2e395f891e7522869be78d7d